common/initial_channel.c

```diff
@@ -58,6 +60,8 @@ struct channel *new_initial_channel(const tal_t *ctx,
 	channel->commitment_number_obscurer
 		= commit_number_obscurer(&channel->basepoints[funder].payment,
 					 &channel->basepoints[!funder].payment);
+	channel->chainparams = chainparams_by_chainhash(chain_hash);
+	assert(channel->chainparams != NULL);
```

Comment:
Per Bolt #2 the receiver must reject the channel if the chain hash is unknown. Do we really just want to assert here?

Reply:
Excellent point, I'm making this an `if (!channel->chainparams) return tal_free(channel);` :+1: